common/functions/src/scalars/expressions/binary.rs

```diff
@@ -17,6 +17,101 @@ use std::marker::PhantomData;
 use common_datavalues2::prelude::*;
 use common_exception::Result;
 
+use super::EvalContext;
+
+pub trait ScalarBinaryFunction2<L: Scalar, R: Scalar, O: Scalar> {
```

Comment:
I think we just introduce `ctx` into `ScalarBinaryFunction` and refactor others using this function is better.

`ScalarBinaryExpression2` & `ScalarBinaryFunction2` can be merged into `ScalarBinaryExpression` & `ScalarBinaryRefFunction`

Reply:
> I think we just introduce `ctx` into `ScalarBinaryFunction` and refactor others using this function is better.
> 
> `ScalarBinaryExpression2` & `ScalarBinaryFunction2` can be merged into `ScalarBinaryExpression` & `ScalarBinaryRefFunction`

Yes, I will merged after add unit and stateless tests